sdk/spring/azure-spring-boot-test-aad/src/test/java/com/azure/test/aad/selenium/AADLoginRunner.java

```diff
@@ -0,0 +1,207 @@
+                   .andThen((app, driver, wait) -> LOGGER.info("Test ===> {}.{}() has finished running.",
+                       Thread.currentThread().getStackTrace()[6].getClassName(),
+                       Thread.currentThread().getStackTrace()[6].getMethodName()))
+                   .run(this.app, this.driver, this.wait);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (this.driver != null) {
+                this.driver.quit();
+            }
+            if (this.app != null) {
+                this.app.close();
+            }
+        }
+    }
+
+    private BrowserCommandWithAppRunner login() {
+        return (app, driver, wait) -> {
+            driver.get(app.root() + "oauth2/authorization/azure");
+            wait.until(presenceOfElementLocated(By.name("loginfmt")))
+                .sendKeys(this.username + Keys.ENTER);
+            Thread.sleep(10000);
+
+            driver.findElement(By.name("passwd"))
+                  .sendKeys(this.password + Keys.ENTER);
+            Thread.sleep(10000);
+
+            driver.findElement(By.cssSelector("input[type='submit']")).click();
+            Thread.sleep(10000);
+        };
+    }
+
+    @FunctionalInterface
+    public interface BrowserCommandWithAppRunner {
+
+        default BrowserCommandWithAppRunner andThen(BrowserCommandWithAppRunner after) {
+            Objects.requireNonNull(after);
+            return (AppRunner app, WebDriver driver, WebDriverWait wait) -> {
+                run(app, driver, wait);
+                after.run(app, driver, wait);
+            };
+        }
+
+        default BrowserCommandWithAppRunner login(BrowserCommandWithAppRunner login) {
+            Objects.requireNonNull(login);
+            return (AppRunner app, WebDriver driver, WebDriverWait wait) -> {
+                login.run(app, driver, wait);
+                run(app, driver, wait);
+            };
+        }
+
+        void run(AppRunner app, WebDriver driver, WebDriverWait wait) throws Exception;
+    }
+
+    public static class AADLoginRunnerConfiguration {
+        private final AppRunner app;
+        private final WebDriver driver;
+        private Consumer<AppRunner> configure;
+
+        private AADLoginRunnerConfiguration(Class<?> appClass) {
+            this.configure = defautlConfigure();
+
+            ChromeOptions options = new ChromeOptions();
+            options.addArguments("--headless");
+            options.addArguments("--incognito", "--no-sandbox", "--disable-dev-shm-usage");
+            this.driver = new ChromeDriver(options);
+
+            this.app = new AppRunner(appClass);
+        }
+
+        public AADLoginRunnerConfiguration configure(Consumer<AppRunner> configure) {
+            this.configure = configure;
+            return this;
+        }
+
+        public AADLoginRunner login(String username, String password) {
+            this.configure.accept(this.app);
```

Comment:
Can we move `this.configure.accept(this.app);` to constructor? Further more, can we delete `this.configure`, just use `app.property(...)` directly? it will make code easier to understand.

Reply:
Furthermore, we can delete `AADLoginRunnerConfiguration`, just use `AADLoginRunner`

Reply:
`EasyTester` can be deleted, too. All information already existed in `AADLoginRunner`.